### DISCUSSION TITLE
Icon Request: Social Media

### DISCUSSION THREAD

**Discussion by mtzrmzia** [tailwindlabs/heroicons#387] [Ideas]
I've seen that in some TWUI examples they use some social media icons, can you include them? Also linkedin and Behance

![image](https://user-images.githubusercontent.com/23019968/82480022-ce981000-9a98-11ea-8d60-403846c7004e.png)

**Comment by aanomm** [comment]
- is the source for these social icons known?

**Comment by mtzrmzia** [comment]
yes.. there you go: [Tailwind Landing](https://tailwindui.com/page-examples/landing-page-01)

**Comment by austriker27** [comment]
Id also love to see social media icons added!

**Comment by deadcoder0904** [comment]
Here's a great alternative https://svgporn.com/

The only con with using it is you have to open SVG & change colors if you don't want to use the original logo colors which are tedious to handle. Idk much SVG so not sure if it's easy or hard.

**Comment by geekish** [comment]
https://simpleicons.org/

**Comment by dillingham** [comment]
Just came here to request these

**Comment by gidea** [comment]
Also looking forward to seeing the heroicons library expanded with social media (using tailwindUI btw, and I need the linkedin icon in the same style as the others, most likely will just self host the svg for it)

**Comment by devops-blue** [comment]
This social medias like Linkedin were not included yet? This should be in every icon library. I'm tailwind customer and we utilized HeroIcons by standard, but the social media icons are very need. Is there any plans to include Linkedin, Instagram, Youtube, Facebook, Twitter, and etc?

**Comment by louishugens** [comment]
still no social media icons?

**Comment by zefir-git** [comment]
You could use https://simpleicons.org — Simple SVG icons of thousands of brands (free and open-source https://github.com/simple-icons/simple-icons). The website also tells you what their primary brand colour is and has a link to logo usage guidelines if present.

I personally don‘t think heroicons needs brand icons since these are usually already provided by the brand as SVG and there isn‘t much you can adapt them to make them fit the rest of the heroicons, i.e. you can‘t edit a company‘s logo just so that it matches your website (at least afaik). And when the brand doesn‘t provide an SVG on their "Brand usage" or similar page, you can often find it online. One good place (if you don‘t want to use simpleicons.org) is Wikipedia. It often has an SVG logo of the brand when you open its page (sometimes it's a PNG rendering and there's a link labelled something like "Show original SVG"). And many other websites also provide brand logos.

> **Note**: To use simpleicons.org with text colour (i.e. `text-blue-500` in TW, etc.) you need to add `fill="currentColor"` to the `<svg>` tag.

**Comment by raphaeltm** [comment]
I'm pretty new to Tailwind, but I bought TailwindUI at a friend's recommendation, and generally I'm really frustrated by the whole ecosystem because of stuff like this. 

* Tailwind UI mixes and matches between Heroicons which is missing a bunch of stuff and randomly providing inline SVGs for other stuff. 
* The site templates use one style of button, but the components look completely different.
* The templates use straight-up colors (i.e. `indigo-500` etc.) all over the place, making it hugely painful to rebrand to your own colours. It should really define hierarchical brand colours (primary/secondary/tertiary etc. with variants/lightnesses).

The whole thing feels so frustrating, especially when I came in with the promise that I would be saving so much time by using this system (Tailwind and/or TailwindUI).

Sorry if I'm a bit rant-y. Tailwind is really neat in many ways 🙂 but overall I've found myself wishing I had stuck to my previous tooling today, given how much time I've been spending dealing with this stuff 😞

**Reply by zefir-git** [reply]
I also mix a lot of Heroicons and also add my own SVGs. I don't think you can expect Heroicons to add *any* possible icon you might need (like the logo of a 1-person company). You could have something like a stone business, but I don't think having an icon for marble, granite and whatever else would be beneficial to the project.

Using colours like `indigo-500` *is* the point of Tailwind. If you want `.primary`, then the next thing you want is `.button`, `.card`, etc. (Although you can actually do that in TW.) I don't think that's the point of TW. You probably don't want to copy a template and simply change the branding. Or copy any component and don't change it at all. You'd pretty much want to customise everything based on your ideas, brand, style, etc. Imagine if all websites looked the same except only with different colours, just because they all found some cheap template.

Anyways, if you want to change e.g. `indigo` to `blue` or `primary`, just do a "replace all in project".

Unlike something like "Bootstrap", Tailwind does not enforce a certain look on your website. You can take your 5th grade CSS school project and rewrite it with tailwind without changing how it looks.

**Reply by raphaeltm** [reply]
Apologies for the rant-y post. I was really tired and frustrated yesterday when I wrote it. I think more and more I'm just realizing that the Tailwind and/or Tailwind UI philosophy isn't a good fit for my specific needs. I'm usually looking for something that I can customize (heavily), rather than having something as a starting point to then rewrite.

I still think my frustration with the icons is fair: the Tailwind team makes Heroicons, and they make Tailwind UI. Tailwind UI uses a combination of Heroicons in some places, and inline SVGs in others. Surely they could have just made those SVGs available in Heroicons, since it's their own project.
